File: day1/cpp/b.cpp
#include <cstdio>
#include <istream>
#include <iterator>
#include <vector>
#include <queue>
#include <stack> 
#include <math.h>
#include <string>
#include <iostream>
#include <fstream>
#include <sstream>
#include <ios>
using namespace std;

void solve(string data) {
  
}

int main() {
  // PREPROCESSING
  ifstream file("t.txt");
  string data  = "";
  int maxCalories = 0, elfPos = 0, ans = -1, curCalories = 0;
  vector<int> cals;

  while(getline(file, data)){
    if(data.empty()) {
      elfPos++;
      if(curCalories > maxCalories){
        ans = elfPos;
        maxCalories = curCalories;
      } 
      cals.push_back(curCalories);
      curCalories = 0;
      continue;
    }
    curCalories+= stoi(data);
  }

  int N = cals.size() -1;
  sort(cals.begin(), cals.end());
  cout << "sum of top three elf's calories " << cals[N] + cals[N-1] + cals[N-2] << '\n';

  return 0;
  
}

File: day1/go/b.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"sort"
	"strconv"
	"strings"
)

func main() {

  contents, err := ioutil.ReadFile("input.txt")
  if err != nil {
    log.Fatal(err.Error())
  }

  data := string(contents)

  solve(data)
}

func solve(s string) {
  calories := strings.Split(s, "\n\n")

  elfCalories := []int{}

  for _ , elfCals := range calories {
    curCalories := 0
    for _, calories := range strings.Split(elfCals, "\n") {
      calorie , _ := strconv.ParseInt(calories, 10, 64)
      curCalories += int(calorie)
    }
    
    elfCalories = append(elfCalories, curCalories)
  }

  sort.Slice(elfCalories, func(x int , y int) bool {
    return elfCalories[x] > elfCalories[y]
  })

  fmt.Printf("the top three elfs have cals: %d\n", elfCalories[0]+ elfCalories[1] + elfCalories[2])
}
